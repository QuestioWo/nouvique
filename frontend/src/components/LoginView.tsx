import React from 'react';

import { Result } from 'neverthrow';
import { Row } from 'react-bootstrap';
import { Container } from 'react-bootstrap';
import { Col } from 'react-bootstrap';
import { RouteComponentProps } from 'react-router';

import { ApiTest, apiTestLink, PageProps, resolveGETCall } from '../utils';

import BasePage from './elements/BasePage';
import LoginForm from './elements/CredentialForm';

import './LoginView.css';

export interface LoginViewProps extends RouteComponentProps, PageProps { }

interface State {
	apiValue: ApiTest
}

export class LoginView extends React.Component<LoginViewProps, State> {
	constructor(props: LoginViewProps) {
		super(props);

		this.state = {
			apiValue: {
				blah: ""
			}
		};
	}

	async componentDidMount() {
		const result: Result<ApiTest, Error> = await resolveGETCall<ApiTest>(apiTestLink);

		result
			.map(res => {
				this.setState({ apiValue: res });

				return null; // necessary to silence warning
			})
			.mapErr(err => {
				console.error(err);
			});
	}

	render() {
		return (
			<React.Fragment>
				<BasePage {...this.props}>


					<Container>
						<Row>

							<Col></Col>

							<Col xs={6}>
								<Row>
									<h1 className="title">
										login
									</h1>
								</Row>

								<Row>
									<div>
										{this.state.apiValue.blah}
									</div>
								</Row>

								<Row>
									<LoginForm {...this.props} />
								</Row>
							</Col>

							<Col></Col>

						</Row>


					</Container>

				</BasePage>
			</React.Fragment>
		);
	}
}