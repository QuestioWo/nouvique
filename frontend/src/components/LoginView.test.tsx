import * as React from 'react';

import { mount, shallow } from 'enzyme';

import { LoginView, LoginViewProps } from './LoginView';
import { MemoryRouter as Router } from 'react-router';
import { loadStripe } from '@stripe/stripe-js';

describe('LoginView', () => {
	const getProps = (): LoginViewProps => (
		{
			history: {} as any,
			location: {} as any,
			match: {} as any,
			updateAlertBar: async () => { },
			emptyBasket: () => { },
			addToBasket: () => { },
			removeFromBasket: () => { },
			getBasketItems: () => [],
			getTotalBasketPrice: () => 0,
			checkBasketAvailabilities: async () => { },
			stripePromise: loadStripe("foo"),
			alert: {} as any
		}
	)

	const getComponent = (props: LoginViewProps) => (
		<LoginView {...props} />
	);


	it('Should render correctly with shallow', () => {
		const component = shallow(
			getComponent(getProps())
		);
		expect(component.debug()).toMatchSnapshot();
	});

	it('Should render correctly with mount', () => {
		// have to include router as will fail with react-router problem due to use of specific functions
		const component = mount(
			<Router>
				{getComponent(getProps())}
			</Router>
		);
		expect(component.debug()).toMatchSnapshot();
	});
});