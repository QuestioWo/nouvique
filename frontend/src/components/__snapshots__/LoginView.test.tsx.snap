// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginView Should render correctly with mount 1`] = `
"<MemoryRouter>
  <Router history={{...}}>
    <LoginView history={{...}} location={{...}} match={{...}} updateAlertBar={[Function: updateAlertBar]} emptyBasket={[Function: emptyBasket]} addToBasket={[Function: addToBasket]} removeFromBasket={[Function: removeFromBasket]} getBasketItems={[Function: getBasketItems]} getTotalBasketPrice={[Function: getTotalBasketPrice]} checkBasketAvailabilities={[Function: checkBasketAvailabilities]} stripePromise={{...}} alert={{...}}>
      <BasePage history={{...}} location={{...}} match={{...}} updateAlertBar={[Function: updateAlertBar]} emptyBasket={[Function: emptyBasket]} addToBasket={[Function: addToBasket]} removeFromBasket={[Function: removeFromBasket]} getBasketItems={[Function: getBasketItems]} getTotalBasketPrice={[Function: getTotalBasketPrice]} checkBasketAvailabilities={[Function: checkBasketAvailabilities]} stripePromise={{...}} alert={{...}}>
        <NavBar>
          <Navbar bg=\\"transparent\\" variant=\\"light\\" expand={true} collapseOnSelect={false}>
            <nav className=\\"navbar navbar-expand navbar-light bg-transparent\\">
              <withRouter(LinkContainer) to=\\"/\\">
                <LinkContainer to=\\"/\\" history={{...}} location={{...}} match={{...}} staticContext={[undefined]} replace={false} exact={false} strict={false} activeClassName=\\"active\\">
                  <Route path=\\"/\\" exact={false} strict={false}>
                    <NavbarBrand className=\\"active\\" style={{...}} href=\\"/\\" onClick={[Function (anonymous)]}>
                      <a style={{...}} href=\\"/\\" onClick={[Function (anonymous)]} className=\\"active navbar-brand\\">
                        <img alt=\\"\\" src=\\"/logo192.png\\" className=\\"nav-bar-logo d-inline-block align-top\\" />
                         
                        Nouvique
                      </a>
                    </NavbarBrand>
                  </Route>
                </LinkContainer>
              </withRouter(LinkContainer)>
              <NavbarCollapse className=\\"justify-content-end\\">
                <ForwardRef in={false} className=\\"justify-content-end\\" timeout={300} mountOnEnter={false} unmountOnExit={false} appear={false} getDimensionValue={[Function: getDefaultDimensionValue]}>
                  <ForwardRef addEndListener={[Function: transitionEndListener]} in={false} timeout={300} mountOnEnter={false} unmountOnExit={false} appear={false} aria-expanded={{...}} onEnter={[Function (anonymous)]} onEntering={[Function (anonymous)]} onEntered={[Function (anonymous)]} onExit={[Function (anonymous)]} onExiting={[Function (anonymous)]} childRef={{...}}>
                    <Transition in={false} timeout={300} mountOnEnter={false} unmountOnExit={false} appear={false} aria-expanded={{...}} onEnter={[Function (anonymous)]} onEntered={[Function (anonymous)]} onEntering={[Function (anonymous)]} onExit={[Function (anonymous)]} onExited={[Function (anonymous)]} onExiting={[Function (anonymous)]} addEndListener={[Function (anonymous)]} nodeRef={{...}} enter={true} exit={true}>
                      <div className=\\"justify-content-end navbar-collapse collapse\\" aria-expanded={{...}}>
                        <Nav className=\\"mr-auto\\" justify={false} fill={false}>
                          <ForwardRef as=\\"div\\" activeKey={[undefined]} className=\\"mr-auto navbar-nav\\" onSelect={[Function (anonymous)]}>
                            <div className=\\"mr-auto navbar-nav\\" onKeyDown={[Function: handleKeyDown]} role={[undefined]}>
                              <NavItem>
                                <div className=\\"nav-item\\">
                                  <NavLink className=\\"nav-link nav-right-side\\" isActive={[Function: isActive]} activeClassName=\\"nav-link active nav-right-side\\" to={{...}}>
                                    <Link aria-current=\\"page\\" className=\\"nav-link nav-right-side nav-link active nav-right-side\\" style={{...}} to={{...}}>
                                      <LinkAnchor aria-current=\\"page\\" className=\\"nav-link nav-right-side nav-link active nav-right-side\\" style={{...}} href=\\"/\\" navigate={[Function: navigate]}>
                                        <a aria-current=\\"page\\" className=\\"nav-link nav-right-side nav-link active nav-right-side\\" style={{...}} href=\\"/\\" onClick={[Function: onClick]}>
                                          Home
                                        </a>
                                      </LinkAnchor>
                                    </Link>
                                  </NavLink>
                                </div>
                              </NavItem>
                              <NavItem>
                                <div className=\\"nav-item\\">
                                  <NavLink className=\\"nav-link nav-right-side\\" activeClassName=\\"nav-link active nav-right-side\\" to={{...}}>
                                    <Link aria-current={{...}} className=\\"nav-link nav-right-side\\" style={[undefined]} to={{...}}>
                                      <LinkAnchor aria-current={{...}} className=\\"nav-link nav-right-side\\" style={[undefined]} href=\\"/media\\" navigate={[Function: navigate]}>
                                        <a aria-current={{...}} className=\\"nav-link nav-right-side\\" style={[undefined]} href=\\"/media\\" onClick={[Function: onClick]}>
                                          Media
                                        </a>
                                      </LinkAnchor>
                                    </Link>
                                  </NavLink>
                                </div>
                              </NavItem>
                              <NavItem>
                                <div className=\\"nav-item\\">
                                  <NavLink className=\\"nav-link nav-right-side\\" isActive={[Function: isActive]} activeClassName=\\"nav-link active nav-right-side\\" to={{...}}>
                                    <Link aria-current={{...}} className=\\"nav-link nav-right-side\\" style={[undefined]} to={{...}}>
                                      <LinkAnchor aria-current={{...}} className=\\"nav-link nav-right-side\\" style={[undefined]} href=\\"/login\\" navigate={[Function: navigate]}>
                                        <a aria-current={{...}} className=\\"nav-link nav-right-side\\" style={[undefined]} href=\\"/login\\" onClick={[Function: onClick]}>
                                          Profile
                                        </a>
                                      </LinkAnchor>
                                    </Link>
                                  </NavLink>
                                </div>
                              </NavItem>
                              <NavItem>
                                <div className=\\"nav-item\\">
                                  <NavLink className=\\"nav-link nav-right-end\\" activeClassName=\\"nav-link active\\" to={{...}}>
                                    <Link aria-current={{...}} className=\\"nav-link nav-right-end\\" style={[undefined]} to={{...}}>
                                      <LinkAnchor aria-current={{...}} className=\\"nav-link nav-right-end\\" style={[undefined]} href=\\"/login\\" navigate={[Function: navigate]}>
                                        <a aria-current={{...}} className=\\"nav-link nav-right-end\\" style={[undefined]} href=\\"/login\\" onClick={[Function: onClick]}>
                                          Login/Signup
                                        </a>
                                      </LinkAnchor>
                                    </Link>
                                  </NavLink>
                                </div>
                              </NavItem>
                            </div>
                          </ForwardRef>
                        </Nav>
                      </div>
                    </Transition>
                  </ForwardRef>
                </ForwardRef>
              </NavbarCollapse>
            </nav>
          </Navbar>
        </NavBar>
        <div className=\\"mainbody\\">
          <AlertBar history={{...}} location={{...}} match={{...}} updateAlertBar={[Function: updateAlertBar]} emptyBasket={[Function: emptyBasket]} addToBasket={[Function: addToBasket]} removeFromBasket={[Function: removeFromBasket]} getBasketItems={[Function: getBasketItems]} getTotalBasketPrice={[Function: getTotalBasketPrice]} checkBasketAvailabilities={[Function: checkBasketAvailabilities]} stripePromise={{...}} alert={{...}}>
            <Alert show={true} variant={[undefined]} onClose={[Function (anonymous)]} dismissible={true} transition={{...}} closeLabel=\\"Close alert\\">
              <Fade unmountOnExit={true} in={true} timeout={300} mountOnEnter={false} appear={false}>
                <ForwardRef addEndListener={[Function: transitionEndListener]} unmountOnExit={true} in={true} timeout={300} mountOnEnter={false} appear={false} onEnter={[Function (anonymous)]} childRef={{...}}>
                  <Transition unmountOnExit={true} in={true} timeout={300} mountOnEnter={false} appear={false} onEnter={[Function (anonymous)]} onEntered={[Function (anonymous)]} onEntering={[Function (anonymous)]} onExit={[Function (anonymous)]} onExited={[Function (anonymous)]} onExiting={[Function (anonymous)]} addEndListener={[Function (anonymous)]} nodeRef={{...}} enter={true} exit={true}>
                    <div role=\\"alert\\" className=\\"fade alert alert-dismissible show\\">
                      <CloseButton onClick={[Function (anonymous)]} aria-label=\\"Close alert\\" variant={[undefined]}>
                        <button type=\\"button\\" className=\\"btn-close\\" onClick={[Function (anonymous)]} aria-label=\\"Close alert\\" />
                      </CloseButton>
                    </div>
                  </Transition>
                </ForwardRef>
              </Fade>
            </Alert>
          </AlertBar>
          <Row>
            <div className=\\"row\\">
              <h1>
                Login
              </h1>
            </div>
          </Row>
          <Row>
            <div className=\\"row\\">
              <div />
            </div>
          </Row>
          <Row>
            <div className=\\"row\\">
              <LoginForm history={{...}} location={{...}} match={{...}} updateAlertBar={[Function: updateAlertBar]} emptyBasket={[Function: emptyBasket]} addToBasket={[Function: addToBasket]} removeFromBasket={[Function: removeFromBasket]} getBasketItems={[Function: getBasketItems]} getTotalBasketPrice={[Function: getTotalBasketPrice]} checkBasketAvailabilities={[Function: checkBasketAvailabilities]} stripePromise={{...}} alert={{...}}>
                <Form onSubmit={[Function (anonymous)]}>
                  <form onSubmit={[Function (anonymous)]} className=\\"\\">
                    <FormGroup className=\\"mb-3\\" controlId=\\"username\\">
                      <div className=\\"mb-3\\">
                        <FormLabel column={false} visuallyHidden={false}>
                          <label className=\\"form-label\\" htmlFor=\\"username\\">
                            Username
                          </label>
                        </FormLabel>
                        <FormControl required={true} type=\\"text\\" placeholder=\\"Username\\" value=\\"\\" onChange={[Function: onChange]}>
                          <input required={true} placeholder=\\"Username\\" value=\\"\\" onChange={[Function: onChange]} type=\\"text\\" size={[undefined]} readOnly={[undefined]} id=\\"username\\" className=\\"form-control\\" />
                        </FormControl>
                      </div>
                    </FormGroup>
                    <FormGroup className=\\"mb-3\\" controlId=\\"password\\">
                      <div className=\\"mb-3\\">
                        <FormLabel column={false} visuallyHidden={false}>
                          <label className=\\"form-label\\" htmlFor=\\"password\\">
                            Password
                          </label>
                        </FormLabel>
                        <FormControl required={true} type=\\"password\\" placeholder=\\"Password\\" value=\\"\\" onChange={[Function: onChange]}>
                          <input required={true} placeholder=\\"Password\\" value=\\"\\" onChange={[Function: onChange]} type=\\"password\\" size={[undefined]} readOnly={[undefined]} id=\\"password\\" className=\\"form-control\\" />
                        </FormControl>
                      </div>
                    </FormGroup>
                    <Row>
                      <div className=\\"row\\">
                        <Col>
                          <div className=\\"col\\">
                            <Button type=\\"submit\\" variant=\\"primary\\" active={false} disabled={false}>
                              <button disabled={false} type=\\"submit\\" className=\\"btn btn-primary\\">
                                Log in
                              </button>
                            </Button>
                          </div>
                        </Col>
                        <Col>
                          <div className=\\"col\\">
                            <Button onClick={[Function: onClick]} variant=\\"outline-primary\\" style={{...}} active={false} disabled={false}>
                              <button onClick={[Function: onClick]} style={{...}} disabled={false} type=\\"button\\" className=\\"btn btn-outline-primary\\">
                                Not signed up?
                              </button>
                            </Button>
                          </div>
                        </Col>
                      </div>
                    </Row>
                  </form>
                </Form>
              </LoginForm>
            </div>
          </Row>
        </div>
      </BasePage>
    </LoginView>
  </Router>
</MemoryRouter>"
`;

exports[`LoginView Should render correctly with shallow 1`] = `
"<Fragment>
  <BasePage history={{...}} location={{...}} match={{...}} updateAlertBar={[Function: updateAlertBar]} emptyBasket={[Function: emptyBasket]} addToBasket={[Function: addToBasket]} removeFromBasket={[Function: removeFromBasket]} getBasketItems={[Function: getBasketItems]} getTotalBasketPrice={[Function: getTotalBasketPrice]} checkBasketAvailabilities={[Function: checkBasketAvailabilities]} stripePromise={{...}} alert={{...}}>
    <Row>
      <h1>
        Login
      </h1>
    </Row>
    <Row>
      <div />
    </Row>
    <Row>
      <LoginForm history={{...}} location={{...}} match={{...}} updateAlertBar={[Function: updateAlertBar]} emptyBasket={[Function: emptyBasket]} addToBasket={[Function: addToBasket]} removeFromBasket={[Function: removeFromBasket]} getBasketItems={[Function: getBasketItems]} getTotalBasketPrice={[Function: getTotalBasketPrice]} checkBasketAvailabilities={[Function: checkBasketAvailabilities]} stripePromise={{...}} alert={{...}} />
    </Row>
  </BasePage>
</Fragment>"
`;
